v2/api/containerinstance/v1beta20211001/container_group_types_gen.go

```diff
@@ -0,0 +1,10183 @@
+	return &EncryptionProperties_StatusARM{}
+}
+
+// PopulateFromARM populates a Kubernetes CRD object from an Azure ARM object
+func (properties *EncryptionProperties_Status) PopulateFromARM(owner genruntime.ArbitraryOwnerReference, armInput interface{}) error {
+	typedInput, ok := armInput.(EncryptionProperties_StatusARM)
+	if !ok {
+		return fmt.Errorf("unexpected type supplied for PopulateFromARM() function. Expected EncryptionProperties_StatusARM, got %T", armInput)
+	}
+
+	// Set property ‘KeyName’:
+	if typedInput.KeyName != nil {
+		keyName := *typedInput.KeyName
+		properties.KeyName = &keyName
+	}
+
+	// Set property ‘KeyVersion’:
+	if typedInput.KeyVersion != nil {
+		keyVersion := *typedInput.KeyVersion
+		properties.KeyVersion = &keyVersion
+	}
+
+	// Set property ‘VaultBaseUrl’:
+	if typedInput.VaultBaseUrl != nil {
+		vaultBaseUrl := *typedInput.VaultBaseUrl
+		properties.VaultBaseUrl = &vaultBaseUrl
+	}
+
+	// No error
+	return nil
+}
+
+// AssignPropertiesFromEncryptionPropertiesStatus populates our EncryptionProperties_Status from the provided source EncryptionProperties_Status
+func (properties *EncryptionProperties_Status) AssignPropertiesFromEncryptionPropertiesStatus(source *v20211001s.EncryptionProperties_Status) error {
+
+	// KeyName
+	properties.KeyName = genruntime.ClonePointerToString(source.KeyName)
+
+	// KeyVersion
+	properties.KeyVersion = genruntime.ClonePointerToString(source.KeyVersion)
+
+	// VaultBaseUrl
+	properties.VaultBaseUrl = genruntime.ClonePointerToString(source.VaultBaseUrl)
+
+	// No error
+	return nil
+}
+
+// AssignPropertiesToEncryptionPropertiesStatus populates the provided destination EncryptionProperties_Status from our EncryptionProperties_Status
+func (properties *EncryptionProperties_Status) AssignPropertiesToEncryptionPropertiesStatus(destination *v20211001s.EncryptionProperties_Status) error {
+	// Create a new property bag
+	propertyBag := genruntime.NewPropertyBag()
+
+	// KeyName
+	destination.KeyName = genruntime.ClonePointerToString(properties.KeyName)
+
+	// KeyVersion
+	destination.KeyVersion = genruntime.ClonePointerToString(properties.KeyVersion)
+
+	// VaultBaseUrl
+	destination.VaultBaseUrl = genruntime.ClonePointerToString(properties.VaultBaseUrl)
+
+	// Update the property bag
+	if len(propertyBag) > 0 {
+		destination.PropertyBag = propertyBag
+	} else {
+		destination.PropertyBag = nil
+	}
+
+	// No error
+	return nil
+}
+
+// Generated from: https://schema.management.azure.com/schemas/2021-10-01/Microsoft.ContainerInstance.json#/definitions/ImageRegistryCredential
+type ImageRegistryCredential struct {
+	// Identity: The identity for the private registry.
+	Identity *string `json:"identity,omitempty"`
+
+	// IdentityUrl: The identity URL for the private registry.
+	IdentityUrl *string `json:"identityUrl,omitempty"`
+
+	// Password: The password for the private registry.
+	Password *string `json:"password,omitempty"`
```

Comment:
This shouldn't be a plain string; it should be a `SecretReference` so the value is pulled from the Kubernetes secret store when required - flag this property using `$isSecret` in the configuration.

Reply:
Good catch and thanks for your advice. I double checked [`ImageRegistryCredentials []ImageRegistryCredential`](https://github.com/majguo/azure-service-operator/blob/feature/container-instance/v2/api/containerinstance/v1beta20211001/container_group_types_gen.go#L1151) and [`Password *string`](https://github.com/majguo/azure-service-operator/blob/feature/container-instance/v2/api/containerinstance/v1beta20211001/container_group_types_gen.go#L4338), `ImageRegistryCredentials` is an **array of struct `ImageRegistryCredential`** which includes field `Password`:
```
spec:
  imageRegistryCredentials:
    - password
      username
``` 

So first I tried to flag `$isSecret` in the configuration file as below, but it didn't generate any `SecretReference` for `Passwod` in type files.
```
      ImageRegistryCredentials:
        Password:
          $isSecret: true
```

Then I tried to flag `$isSecret` for type `ImageRegistryCredential` instead of field name `ImageRegistryCredentials`:
```
      ImageRegistryCredential:
        Password:
          $isSecret: true
```
It did generate `SecretReference` for `Passwod` in type files but failed with the following error message:
```
[controller:test] internal/controllers/controller_resources_gen.go:1097:39: obj.Spec.ImageRegistryCredentials.Password undefined (type []"github.com/Azure/azure-service-operator/v2/api/containerinstance/v1beta20211001storage".ImageRegistryCredential has no field or method Password)
```
The error is reasonable as `ImageRegistryCredentials` is an array instead of struct.

So it looks like that `$isSecret` hasn't support for array yet?

Reply:
Yes that's possible. I will take a look at this today and see what I can do to fix it so that it works.

Reply:
Thanks @matthchr for looking into the issue and the fix.

I'm not from the ContainerInstance team, so pls feel free to go ahead and send a PR to fix their Swagger issue you mentioned above.

Reply:
Status update: So I have a fix and that fix is mostly tested, but it wont be in PR until tomorrow morning.

Reply:
@matthchr I tried in commit eb83fce and your fix works as expected, thanks for your help! Cc @theunrepentantgeek .